linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2AccessLogger.scala

```diff
@@ -0,0 +1,84 @@
+package io.buoyant.linkerd.protocol.h2
+
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack, Stackable}
+import com.twitter.finagle.buoyant.h2.{Request, Response}
+import com.twitter.finagle.context.RemoteInfo
+import com.twitter.logging._
+import com.twitter.util.{Time, TimeFormat}
+
+case class H2AccessLogger(log: Logger) extends SimpleFilter[Request, Response] {
+
+  def apply(req: Request, svc: Service[Request, Response]) = {
+    val reqHeaders = req.headers
+    val remoteHost = RemoteInfo.Upstream.addr.getOrElse("-")
```

Comment:
To avoid printing the port here I think you want to attempt to cast to `InetSocketAddress` and use `getHostString`.  Something like

```
RemoteInfo.Upstream.addr match {
  case Some(isa: InetSocketAddress) => isa.getHostString
  case _ => "-"
}
```

Reply:
awesome, thanks @adleong! i had to import `InetSocketAddress` and hope I choose the correct one?!